Topics/Software_Engineering/Ethics_In_Software_Engineering.md

```diff
@@ -0,0 +1,112 @@
+
+### Professionalism
+As a software engineer, you are personally and professionally responsible for maintaining a high level of quality in your work. The same applies to the level of transparency and privacy that you provide for your users. As you work with clients, you accept both implicit and explicit terms that include maintining confidentiality and privacy in order to protect their personally identifiable information (PII) and intellectual property.
+
+### Legal Consequences
+There are great legal implications to breaking ethical laws, and being unaware of them is seldom a reasonable justification. Violating the privacy of a user or otherwise handling their information irresponsibly can lead to lawsuits from that user, or from a client company. Poor practices can also result in detrimental fines and a ruined record; in the European Union, for example, the General Data Protection Regulation (GDPR) is strongly enforced, and non-compliance can result in a fine of millions of dollars. The Canadian equivalent to the GDPR is PIPEDA.
+
+## PIPEDA
+Canada's Personal Information Protection and Electronic Documents Act (PIPEDA) is a set of laws regarding data collection and use in the context of businesses. To summarize, here is what it requires:
+
+* Consent: Organizations must obtain consent when collecting, using, or disclosing personal information.
+* Purpose Limitation: Organizations must collect personal information for a specific purpose and cannot use it for other reasons without consent.
+* Limited Collection: Organizations must only collect information necessary for the identified purpose.
+* Accuracy: Organizations must ensure that the collected information is accurate, complete and up-to-date.
+* Safeguards: Organizations must implement security measures to protect personal information against unauthorized access, disclosure, or misuse.
+* Openness: Organizations must be transparent about their policies and practices related to the handling of personal information.
+* Individual Access: Individuals have the right to access their own personal information held by an organization and request corrections if necessary.
+* Challenging Compliance: Individuals can challenge an organization's compliance with PIPEDA and seek recourse.
+
+In general, when designing software that collects user data, you must ensure that you only collect data that users consent to being collected, and only use it for the reasons that they consented to. Full transparency is also required, as users have a right to know which data you are storing if it pertains to them.
+
+## Case Studies
+When evaluating the ethical quality of a service, we can look at three main factors: privacy, consent and security. 
```

Comment:
In amazon's example, I saw you mentioned Biased Algorithm, it would be better if you also add it here

Reply:
I see how that was confusing. I have added accuracy to the list and changed the subheading from **Biased Algorithm** to **Violation of Accuracy* to better fit with the other case studies.

Reply:
yeah, that makes sense to me